Added Interval>>sort: method that raises a "you can not sort an inter…

dupriezt opened the pull request:
…val" error

From fogbuz issue:
Interval new sort raises an "#< was sent to nil" exception
(1 to: 5)  sort raises an "you can not store into an interval" exception

The Interval class inherits the default #sort: method of SequenceableCollection, which does not work for intervals. I am not sure the concept of sorting an Interval makes sense in the general case. I suggest implementing Interval>>sort: so that it raises an explicit exception: "you can not sort an interval" akin to what Interval>>at:put: does (it raises "you can not store into an interval"). Unless someone has an idea for what "sorting an Interval" could mean.

Commit:
Added Interval>>sort: method that raises a "you can not sort an interval" error

juliendelplanque commented on src/Collections-Sequenceable/Interval.class.st:
```diff
@@ -369,6 +369,13 @@ Interval >> size [
+Interval >> sort: aBlock [
+	"What sorting an Interval means is not clear."
+
+	self error: 'you can not sort an interval'
```
Except for the missing capital letter at the beginning of the error message and the dot at the end, this is fine for me. :-)

dupriezt replied:
I wrote it like that to be consistent with the Interval>>at:put: method:
```
at: anInteger put: anObject 
	"Storing into an Interval is not allowed."

	self error: 'you can not store into an interval'
```

dupriezt replied:
But sure, why not.

Commit:
- Error message of Interval>>sort: now starts with a capital letter and ends with a dot.
- Interval>>sort: method moved to sorting protocol so that it is in the same protocol as the sort: method of the superclass

pavel-krivanek commented on src/Collections-Sequenceable/Interval.class.st:
```diff
@@ -369,6 +369,13 @@ Interval >> size [
+Interval >> sort: aBlock [
+	"What sorting an Interval means is not clear."
+
+	self error: 'You can not sort an interval.'
```
we should use `ShouldNotImplement` exception instead of error

dupriezt replied:
Thanks. I made a new commit replacing this `self error:` with `self shouldNotImplement`

MarcusDenker commented:
the case on the issue tracker is https://pharo.fogbugz.com/f/cases/22588/Sorting-an-Interval-raises-an-exception

Commit:
Using self shouldNotImplement in Interval>>#sort instead of raising a generic error

dupriezt commented:
@MarcusDenker @pavel-krivanek 
Added a new commit to follow Pavel's suggestion of using `self shouldNotImplement` instead of `self error:`
